### ISSUE TITLE
Doesn't work with fields which are not primary or foreign keys

### ISSUE THREAD

**Issue by lurodrigo** [sloanelybutsurely/typeid-elixir#37]
Using `TypeID` with regular fields (not primary keys or assocs) doesn't work.

Using something like `field :external_id, TypeID, prefix: "venue", type: :uuid` doesn't work. After a bit of investigation, the reason was the following lines in `lib/type_id/ecto.ex`:

```elixir
      if primary_key do
        %{primary_key: primary_key, schema: schema, field: field, prefix: prefix, type: type}
      else
        %{schema: schema, field: field, type: type}
      end
    end

    defp find_prefix(%{prefix: prefix}), do: prefix

    defp find_prefix(%{schema: schema, field: field}) do
      %{related: schema, related_key: field} = schema.__schema__(:association, field)

      prefix =
        case schema.__schema__(:type, field) do
          {:parameterized, {TypeID, %{prefix: prefix}}} -> prefix
          {:parameterized, TypeID, %{prefix: prefix}} -> prefix
          _ -> nil
        end

      prefix
    end
```

It seems to assume all fields are either primary keys, for which we read the `prefix` from the map directly, or assocs, for which we lookup the `prefix` in the assoc's module. 

I made a PR with the minimal changes https://github.com/lurodrigo/typeid-elixir/pull/1/files to make it work for me (2 lines). But I suspect a better approach would be to, instead of using a boolean `primary_key` field, use a `key_type :: :primary_key | :foreign_key | :none` field and pattern match on that. I suspect such an approach would need to also take the problems described in #36 into consideration. Would such a PR be welcome?

**Comment by sloanelybutsurely** [comment]
I would prefer a PR that just adds support manually specifying the prefix on non-primary key fields. I don't want to change the existing `primary_key` boolean option.

**Comment by PatrikStenmark** [comment]
I just ran in to this issue as well. 

> I would prefer a PR that just adds support manually specifying the prefix on non-primary key fields. I don't want to change the existing primary_key boolean option.

Isn't this what @lurodrigo have implemented in his PR?

**Comment by sloanelybutsurely** [comment]
fixed by #39
